azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventHubClient.java

```diff
@@ -53,7 +53,21 @@ private EventHubClient(final ConnectionStringBuilder connectionString) throws IO
      */
     public static EventHubClient createFromConnectionStringSync(final String connectionString)
             throws EventHubException, IOException {
-        return createFromConnectionStringSync(connectionString, null);
+        return createFromConnectionStringSync(connectionString, null, ClientInfo.Default);
+    }
+
+    /**
+     * Synchronous version of {@link #createFromConnectionString(String)}.
+     *
+     * @param connectionString The connection string to be used. See {@link ConnectionStringBuilder} to construct a connectionString.
+     * @param clientInfo       A custom {@link ClientInfo} to be sent to EventHub
+     * @return EventHubClient which can be used to create Senders and Receivers to EventHub
+     * @throws EventHubException If Service Bus service encountered problems during connection creation.
+     * @throws IOException         If the underlying Proton-J layer encounter network errors.
+     */
+    public static EventHubClient createFromConnectionStringSync(final String connectionString, final ClientInfo clientInfo)
```

Comment:
adding the ClientInfo to this façade method of our client might make us look complicated - with too many options for the starters (and then, we will also need to replicate this across all of our clients):

public static EventHubClient createFromConnectionStringSync(final String connectionString, final ClientInfo clientInfo) 

let me discuss with our pack of client developers and come back on the best way to pass this.